src/rgw/rgw_auth.cc

```diff
@@ -508,3 +508,44 @@ RGWAuthApplier::aplptr_t RGWKeystoneAuthEngine::authenticate() const
 
   return nullptr;
 }
+
+rgw::auth::Engine::result_t
+rgw::auth::Strategy::authenticate(const req_state* const s) const
+{
+  for (const stack_item_t& kv : auth_stack) {
+    const rgw::auth::Engine& engine = kv.first;
+    const auto& policy = kv.second;
+
+    auto&& res = engine.authenticate(s);
```

Comment:
I'm trying to figure this out and it looks that making confusion is the sole outcome from requesting the R-value reference here. :-( Undoubtedly needs to be reworked. Thanks for pointing this out.